flytekit/core/python_auto_container.py

```diff
@@ -0,0 +1,275 @@
+        :param task_type: String task type to be associated with this Task
+        :param container_image: String FQN for the image.
+        :param requests: custom resource request settings.
+        :param limits: custom resource limit settings.
+        :param environment: Environment variables you want the task to have when run.
+        :param task_resolver: Custom resolver - will pick up the default resolver if empty, or the resolver set
+          in the compilation context if one is set.
+        """
+        super().__init__(
+            task_type=task_type, name=name, task_config=task_config, **kwargs,
+        )
+        self._container_image = container_image
+        # TODO(katrogan): Implement resource overrides
+        self._resources = ResourceSpec(
+            requests=requests if requests else Resources(), limits=limits if limits else Resources()
+        )
+        self._environment = environment
+
+        compilation_state = FlyteContext.current_context().compilation_state
+        if compilation_state and compilation_state.task_resolver:
+            if task_resolver:
+                logger.info(
+                    f"Not using the passed in task resolver {task_resolver} because one found in compilation context"
+                )
+            self._task_resolver = compilation_state.task_resolver
+            if self._task_resolver.task_name(self) is not None:
+                self._name = self._task_resolver.task_name(self)
+        else:
+            self._task_resolver = task_resolver or default_task_resolver
+
+    @property
+    def task_resolver(self) -> TaskResolverMixin:
+        return self._task_resolver
+
+    @property
+    def container_image(self) -> Optional[str]:
+        return self._container_image
+
+    @property
+    def resources(self) -> ResourceSpec:
+        return self._resources
+
+    @abstractmethod
+    def get_command(self, settings: SerializationSettings) -> List[str]:
+        pass
+
+    def get_container(self, settings: SerializationSettings) -> _task_model.Container:
+        env = {**settings.env, **self.environment} if self.environment else settings.env
+        return _get_container_definition(
+            image=get_registerable_container_image(self.container_image, settings.image_config),
+            command=[],
+            args=self.get_command(settings=settings),
+            data_loading_config=None,
+            environment=env,
+            storage_request=self.resources.requests.storage,
+            cpu_request=self.resources.requests.cpu,
+            gpu_request=self.resources.requests.gpu,
+            memory_request=self.resources.requests.mem,
+            storage_limit=self.resources.limits.storage,
+            cpu_limit=self.resources.limits.cpu,
+            gpu_limit=self.resources.limits.gpu,
+            memory_limit=self.resources.limits.mem,
+        )
+
+
+class TaskResolverMixin(object):
```

Comment:
python 3+: `object` is default parent class

```suggestion
class TaskResolverMixin
```

Reply:
i like keeping it though, makes it explicit and it's everywhere else.